x/dex/keeper/msgserver/msg_server_contract_deposit_rent.go

```diff
@@ -0,0 +1,37 @@
+package msgserver
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	appparams "github.com/sei-protocol/sei-chain/app/params"
+	"github.com/sei-protocol/sei-chain/x/dex/types"
+)
+
+func (k msgServer) ContractDepositRent(goCtx context.Context, msg *types.MsgContractDepositRent) (*types.MsgContractDepositRentResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	contract, err := k.GetContract(ctx, msg.ContractAddr)
+	if err != nil {
+		return nil, err
+	}
+	if contract.RentBalance > 0 && contract.Creator != msg.Sender {
```

Comment:
why don't we just use whether the creator == sender to check here, and let people to deposit whenever they want

Reply:
just concerned about legacy contracts which doesn't have a creator set. We don't want multiple people to deposit to it because only one can withdraw

Reply:
think there is a chance that an attack can take the creator role if the contract isn't registered with rent